app/src/main/java/org/wikipedia/analytics/eventplatform/SamplingConfig.kt

```diff
@@ -0,0 +1,38 @@
+package org.wikipedia.analytics.eventplatform
+
+import androidx.annotation.VisibleForTesting
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Represents the sampling config component of a stream configuration.
+ *
+ * The boxed Double type is used instead of the double primitive because its value may be null,
+ * which denotes that the stream should always be *included*.
+ */
+ class SamplingConfig {
+
+    enum class Identifier {
+        @SerializedName("pageview")
+        PAGEVIEW,
+        @SerializedName("session")
+        SESSION,
+        @SerializedName("device")
+        DEVICE
+    }
+
+    private var identifier: Identifier? = null
+    var rate = 1.0
+
+    // This constructor is needed for correct Gson deserialization. Do not remove!
+    constructor() {}
```

Comment:
Not sure but would it be possible to have this and the `identifier` and `rate` in the constructor as following:
```
class SamplingConfig(var rate: Double = 1.0,
                      var identifier: Identifier = Identifier.SESSION) {
    ...
}
```

Reply:
Have left it as it because of the comment .... not so easy to test this as it is sampling config for MEP ... so can we leave it as is?

Reply:
Makes sense to me.